### PR TITLE
Exclude non-API elements from javadoc

### PR DESCRIPTION
**Pull request by anistor**
https://issues.redhat.com/browse/ISPN-11043

### CODE REVIEW

**Comment by anistor** [comment]
This will obviously fail now because it depends on the doclet ver. 1.2.0.Final to be released first.
See https://github.com/infinispan/infinispan-doclets/pull/1

**Comment by anistor** [comment]
It's odd we have some ```@public``` classes in ```org.infinispan.security.impl``` package @tristantarrant

**Comment by tristantarrant** [comment]
Right, those are the role mappers which should probably be moved one level up.

**Comment by tristantarrant** [comment]
@anistor https://github.com/infinispan/infinispan/pull/7655

File: build-configuration/pom.xml [hunk 1]
```diff
@@ -145,7 +145,7 @@
       <version.hibernate.search.module.slot>5.10</version.hibernate.search.module.slot>
       <version.infinispan>${project.version}</version.infinispan>
       <version.infinispan.arquillian>1.2.0.Beta3</version.infinispan.arquillian>
-      <version.infinispan.doclets>1.1.0.Final</version.infinispan.doclets>
+      <version.infinispan.doclets>1.2.0-SNAPSHOT</version.infinispan.doclets>
```
Comment (tristantarrant) [inline build-configuration/pom.xml]: 1.2.1.Final please

Comment (anistor) [reply build-configuration/pom.xml]: done

**Comment by anistor** [comment]
Rebased ,  @tristantarrant

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**